lib/edit-site-page.php

```diff
@@ -143,11 +152,31 @@ function gutenberg_edit_site_init( $hook ) {
 		)
 	);
 
-	// Preload server-registered block schemas.
 	wp_add_inline_script(
 		'wp-blocks',
-		'wp.blocks.unstable__bootstrapServerSideBlockDefinitions(' . wp_json_encode( get_block_editor_server_block_settings() ) . ');'
```

Comment:
Is this not needed anymore?

Reply:
Yeah, I have the same question.

Reply:
This isn't just a removal, but a change from this:

`wp.blocks.unstable__bootstrapServerSideBlockDefinitions(' . wp_json_encode( get_block_editor_server_block_settings() ) . ');`

 to this:
`sprintf( 'wp.blocks.setCategories( %s );', wp_json_encode( get_block_categories( $post ) ) ),`

Which matches how WP core does it. 

Bringing that back brings the custom blocks back. That line was added here:
https://github.com/WordPress/gutenberg/commit/d3d174797ece09fad7e327a84abd8809e98c4577

But I am not sure why it (unstable__bootstrapServerSideBlockDefinitions) was used over the standard one in core (wp.blocks.setCategories).

Reply:
That being said, nothing appears to break by this change, but it does make custom blocks appear in the inserter while in `Site Edit` mode.

Reply:
cc @gziolo

Reply:
I think you need both, `wp.block.setCategories` registers block categories that are filterable on the server. If someone uses PHP to register/unregister those categories on the server, this change wouldn't be propagated to the client. As far as I remember, by removing a given category, you would prevent blocks registration on the client for blocks that define it.


I think `wp.blocks.unstable__bootstrapServerSideBlockDefinitions` has also some valid usage. We want to make it possible to share definitions between the server and client but it isn't the case for all block as of today. @aduth might also better remember why it works this way, but I think that some blocks register attributes only on the server so this helps to propagate them to the client. I see usage in WordPress core:
https://github.com/WordPress/wordpress-develop/blob/437ac63d7652b08095dc7f77b6ba4ef52db44dbb/src/wp-admin/edit-form-blocks.php#L111-L115

Reply:
Ah ok! I will just add this one in addition. Thanks @gziolo

Reply:
> I think `wp.blocks.unstable__bootstrapServerSideBlockDefinitions` has also some valid usage. We want to make it possible to share definitions between the server and client but it isn't the case for all block as of today. @aduth might also better remember why it works this way, but I think that some blocks register attributes only on the server so this helps to propagate them to the client. I see usage in WordPress core:
> https://github.com/WordPress/wordpress-develop/blob/437ac63d7652b08095dc7f77b6ba4ef52db44dbb/src/wp-admin/edit-form-blocks.php#L111-L115

Yes, without this, the client editor would not reflect any block settings which were registered server-side using `register_block_type`. I'm not sure how widely it is used, but it is expected that there be a "merge" of settings defined server-side and client-side.